indra/newview/scripts/lua/require/UI.lua

```diff
@@ -122,4 +122,19 @@ function UI.type(...)
     end
 end
 
+-- ***************************************************************************
+--  Snapshot
+-- ***************************************************************************
+-- UI.snapshot{filename=filename            -- extension may be specified: bmp, jpeg, png
+--          [, type='COLOR' | 'DEPTH']
+--          [, width=width][, height=height]  -- uses current window size if not specified
+--          [, showui=true][, showhud=true]
+--          [, rebuild=false]}
+function UI.snapshot(...)
+    local args = mapargs('filename,width,height,showui,showhud,rebuild,type', ...)
+    leap.request('LLViewerWindow', {op='saveSnapshot', filename = args.filename,
+                                    width=args.width, height=args.height,
+                                    showui=args.showui, showhud=args.showhud,
+                                    rebuild=args.rebuild, type=args.type})
```

Comment:
For passing all these keys as-is, it would be simpler to write:
```
local args = etc.
args.op = 'saveSnapshot'
return leap.request('LLViewerWindow', args).result  -- or ok, as the case may be
```